pandas/core/base.py

```diff
@@ -677,19 +676,20 @@ def item(self):
         """
         Return the first element of the underlying data as a python scalar.
 
-        .. deprecated:: 0.25.0
-
         Returns
         -------
         scalar
             The first element of %(klass)s.
+
+        Raises
+        ------
+        ValueError
+            If the data is not length-1.
         """
-        warnings.warn(
-            "`item` has been deprecated and will be removed in a future version",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return self.values.item()
+        if len(self) == 1:
```

Comment:
Was adding this condition discussed somewhere? I would have thought just keep existing behaviour

Reply:
This is the bugfix part.  dt64, dt64tz, and td64 we're currently incorrectly returning int

Reply:
Hmm doesn't this break non-DTA though? 

```python
>>> type(pd.Series(range(1)).item())
<class 'int'>
>>> type(pd.Series(range(1))[0])
<class 'numpy.int64'>
```

I thought one of the points of item was to return a Python object (at least in the Numpy world)

Reply:
^ current behavior

Reply:
Yes, I think we should keep the behaviour of `item` to return a python scalar (where possible of course, so for datetime/timedelta it is fine to return a pandas Timestamp/Timedelta I think)

Reply:
ok, will update.

Reply:
Has this been resolved?

Reply:
I think the concerns raised by @WillAyd and @jorisvandenbossche have been addressed